erpnext/public/js/controllers/transaction.js

```diff
@@ -1700,6 +1700,29 @@ erpnext.TransactionController = erpnext.taxes_and_totals.extend({
 		}
 	},
 
+	set_query_for_item_tax_template: function(doc, cdt, cdn) {
+
+		var item = frappe.get_doc(cdt, cdn);
+		if(!item.item_code) {
+			frappe.throw(__("Please enter Item Code to get item taxes"));
```

Comment:
Purchase and Sales Invoice lines should never need an item code for anything, to support one-off manually typed lines